Testscripts/Windows/PERF-XDP-NTTTCP.ps1

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+<#
+.Description
+    This script deploys the VM and Verifies there are no regression in Network Performance due
+    to XDP by comparing ntttcp results.
+#>
+
+param([object] $AllVmData,
+    [object] $CurrentTestData)
+
+$iFaceName = "eth1"
+$thresholdThroughput = 0.90
+
+function Run_NTTTCP_PERF {
+    $ResultDir = $args[0]
+    Write-LogInfo "Starting ntttcp perf test"
+    $testJob = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+        -username $user -password $password -command "./perf_ntttcp.sh &> ~/ntttcpConsoleLogs.txt" `
+        -RunInBackground -runAsSudo
+    while ((Get-Job -Id $testJob).State -eq "Running") {
+        $currentStatus = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -username $user -password $password -command "tail -2 ~/ntttcpConsoleLogs.txt | head -1" -runAsSudo
+        Write-LogInfo "Current Test Status: $currentStatus"
+        Wait-Time -seconds 20
+    }
+    # Copy result
+    Copy-RemoteFiles -downloadFrom $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+        -username $user -password $password -download `
+        -downloadTo $ResultDir -files "~/ntttcp-tcp-test-logs/report.*" -runAsSudo
+}
+
+function Compare_NTTTCP_Result {
+    $beforeDirPath = Join-Path -Path $args[0] -ChildPath 'report.csv'
+    $afterDirPath = Join-Path -Path $args[1] -ChildPath 'report.csv'
+
+    Write-LogInfo "Importing CSV from path $afterDirPath"
+    $csvWithXDP = Import-Csv -Path $afterDirPath
+    Write-LogInfo "Importing CSV from path $beforePath"
+    $csvWithoutXDP = Import-Csv -Path $beforeDirPath
+    $testFailedFlag = $false
+    if ( ($csvWithXDP).Count -ne ($csvWithoutXDP).Count) {
+        Throw "NTTTCP did not executed properly. Different num lines in report.csv"

```

Comment:
Please add counts data. Think if you have the test failure and read the log. The logging should be helpful.

Reply:
Fixed in: 52245c7dce4934e96ef35dcc4c6aa36315255e62